taraxa/state/state_transition/state_transition.go

```diff
@@ -102,36 +103,56 @@ func (self *StateTransition) evm_state_checkpoint() {
 	self.evm_state.CommitTransaction(&self.trie_sink, self.evm.GetRules().IsEIP158)
 }
 
-func (self *StateTransition) BeginBlock(blk_info *vm.BlockInfo, rewards map[common.Address]*big.Int) {
+func (self *StateTransition) BeginBlock(blk_info *vm.BlockInfo) {
 	self.begin_block()
 	blk_n := self.pending_blk_state.GetNumber()
 	rules_changed := self.evm.SetBlock(&vm.Block{blk_n, *blk_info}, self.chain_config.ETHChainConfig.Rules(blk_n))
 	if self.dpos_contract != nil && rules_changed {
 		self.dpos_contract.Register(self.evm.RegisterPrecompiledContract)
 	}
-	if self.dpos_contract != nil {
-		self.dpos_contract.BeginBlockCall(rewards)
-	}
 	if self.chain_config.ETHChainConfig.IsDAOFork(blk_n) {
 		misc.ApplyDAOHardFork(&self.evm_state)
 		self.evm_state_checkpoint()
 	}
 }
 
-func (self *StateTransition) ExecuteTransaction(trx *vm.Transaction) (ret vm.ExecutionResult) {
-	ret = self.evm.Main(trx, self.chain_config.ExecutionOptions)
+func (self *StateTransition) ExecuteTransaction(tx *vm.Transaction) (ret vm.ExecutionResult) {
+	ret = self.evm.Main(tx, self.chain_config.ExecutionOptions)
 	self.evm_state_checkpoint()
 	return
 }
 
-func (self *StateTransition) EndBlock(uncles []state_common.UncleBlock) {
+func (self *StateTransition) GetChainConfig() (ret *chain_config.ChainConfig) {
+	ret = self.chain_config
+	return
+}
+
+func (self *StateTransition) EndBlock(uncles []state_common.UncleBlock, rewardsStats *rewards_stats.RewardsStats, rewardsPayouts map[common.Address]*big.Int) {
 	if !self.chain_config.DisableBlockRewards {
 		evm_block := self.evm.GetBlock()
-		ethash.AccumulateRewards(
-			self.evm.GetRules(),
-			ethash.BlockNumAndCoinbase{evm_block.Number, evm_block.Author},
-			uncles,
-			&self.evm_state)
+
+		if self.chain_config.ExecutionOptions.DisableStatsRewards {
+			ethash.AccumulateRewards(
+				self.evm.GetRules(),
+				ethash.BlockNumAndCoinbase{evm_block.Number, evm_block.Author},
+				uncles,
+				rewardsPayouts,
+				&self.evm_state)
+		} else {
+			if self.dpos_contract == nil {
+				panic("Stats rewards enabled but no dpos contract registered")
```

Comment:
this should not be there as there are some tests without dpos_contract inside the EVM

Reply:
those have to set disable_rewards_stats flag